src/main/java/seedu/duke/food/Food.java

```diff
@@ -332,12 +332,12 @@ public String toString()  {
             Ui.showError("Expiry date parsing error");
         }
 
-        if (!Objects.equals(unit, "")) {
+        if (quantity != 0.0) {
```

Comment:
I would aggree changing this to this as well, as there will always be a default unit as "UNIT" or "UNITS" depending on the quantity